docs/cosa/buildextend-secex.md

```diff
@@ -0,0 +1,40 @@
+---
+parent: CoreOS Assembler Command Line Reference
+nav_order: 1
+---
+
+# cosa buildextend-secex
+
+This buildextend command is used to build Secure Execution enabled QEMU-Images for IBM Z.
+In order to build a Secure Execution protected QEMU-Image, you need to provide a Hostkey to encrypt it.
+
+For more information on Secure Execution on IBM Z, please refer to the [IBM Documentation](https://www.ibm.com/docs/en/linux-on-systems?topic=ibmz-secure-execution).
+
+The command is intended to be used in the RHCOS CI together with the Universal Hostkey, such that the Image can be booted up on any IBM Z Machine that supports Secure Execution.
+This results in a few specifics to note:
+- The resulting image will only be encrypted with a single Hostkey, to enable firstboot
+- The Hostkey will not be written to the image
+- The Hostkey(s) need to be provided later during firstboot through ignition
+    - Firstboot service will fail when no Hostkey is provided, as the sdboot-image can not be recreated
+    - The Hostkey(s) should be written to `/etc/se-hostkeys/ibm-z-hostkey-<hostkey-name>.crt`
+
+To facilitate this, `buildextend-secex` can take 2 mutually exclusive additional arguments: `--genprotimgvm <path>` and `--hostkey <path>`.
+If none is provided, `--genprotimgvm` is used with default values.
+
+## `--genprotimgvm <path>` (default)
+
+Default Value: `/data.secex/genprotimgvm.qcow2`
+
+This path is the default behavior. It assumes that the Hostkey is not directly available, but is instead only supplied through a Secure Execution protected VM.
+
+The QEMU-Image will be build normally, however it will not run `genprotim` or `zipl`, but instead save the required input for the command to a temporary location.
+After the build, it will run the provided VM. The VM is used to isolate and protect the `genprotimg` command, such that the Universal Hostkey will not be exposed.
+It will call a provided bash script before and after the genprotimg step.
+This enables us to copy the required kernel, initramfs and parmfile to the VM and afterwards move the sdboot-image to the disk, as well as calling `zipl` to make the image bootable.
+
+## `--hostkey <path>`
+
+This path is intended for local development, but can be used for custom builds. It takes the path to a single Hostkey and uses it to build the image.
+
+Instead of running `genprotimg` and `zipl` in a separate VM, they will be run during the build process. Otherwise the build is identical to the `--genprotimgvm`.
```

Comment:
```suggestion
Instead of running `genprotimg` and `zipl` in a separate VM, they run during the build process. Otherwise, the build is identical to the `--genprotimgvm`.
```